sdk/containerregistry/container-registry/src/utils/helpers.ts

```diff
@@ -29,3 +29,31 @@ export async function readStreamToEnd(stream: NodeJS.ReadableStream): Promise<Bu
     stream.on("error", (err) => reject(err));
   });
 }
+
+export async function* readChunksFromStream(
+  stream: NodeJS.ReadableStream,
+  chunkSize: number
+): AsyncGenerator<Buffer> {
+  let chunk = Buffer.alloc(chunkSize);
```

Comment:
We could optimize for memory by not allocating the full chunk initially, in the case that the blob is smaller than the chunk size.

Reply:
Since there's no way of telling the length of the stream in advance, doing this complicates the algorithm substantially. I'd like to take a look at this later on.